plugins/ui/src/deephaven/ui/components/illustrated_message.py

```diff
@@ -0,0 +1,160 @@
+        grid_row: When used in a grid layout, specifies the row the element should be placed in within the grid.
+        grid_column: When used in a grid layout, specifies the column the element should be placed in within the grid.
+        grid_row_start: When used in a grid layout, specifies the starting row to span within the grid.
+        grid_row_end: When used in a grid layout, specifies the ending row to span within the grid.
+        grid_column_start: When used in a grid layout, specifies the starting column to span within the grid.
+        grid_column_end: When used in a grid layout, specifies the ending column to span within the grid.
+        margin: The margin for all four sides of the element.
+        margin_top: The margin for the top side of the element.
+        margin_bottom: The margin for the bottom side of the element.
+        margin_start: The margin for the logical start side of the element, depending on layout direction.
+        margin_end: The margin for the logical end side of the element, depending on layout direction.
+        margin_x: The margin for the left and right sides of the element.
+        margin_y: The margin for the top and bottom sides of the element.
+        width: The width of the element.
+        height: The height of the element.
+        min_width: The minimum width of the element.
+        min_height: The minimum height of the element.
+        max_width: The maximum width of the element.
+        max_height: The maximum height of the element.
+        position: Specifies how the element is position.
+        top: The top position of the element.
+        bottom: The bottom position of the element.
+        left: The left position of the element.
+        right: The right position of the element.
+        start: The logical start position of the element, depending on layout direction.
+        end: The logical end position of the element, depending on layout direction.
+        z_index: The stacking order for the element
+        is_hidden: Hides the element.
+        id: The unique identifier of the element.
+        UNSAFE_class_name: Set the CSS className for the element. Only use as a last resort. Use style props instead.
+        UNSAFE_style: Set the inline style for the element. Only use as a last resort. Use style props instead.
+
+    Examples:
+        prompt = ui.illustrated_message(
+            ui.heading("Enter URL above"),
+            ui.content("Enter a URL of a CSV above and click 'Load' to load it"),
+        )
+        warning = ui.illustrated_message(
+            ui.icon("vsWarning"),
+            ui.heading("Error loading table"),
+            ui.content(f"{error}"),
+        )
+        error_message = ui.illustrated_message(
+            ui.icon("vsWarning", style={"fontSize": "48px"}),
+            ui.heading("Invalid Input"),
+            ui.content("Please enter 'Sym' and 'Exchange' above"),
+        )
```

Comment:
Drop this example... we need to wire up the props in `IconElementView.tsx` correctly, it should be:
```
    // eslint-disable-next-line react/jsx-props-no-spreading
    <Icon {...props}>
      <FontAwesomeIcon icon={icon} />
    </Icon>
```
Instead of passing the props into `FontAwesomeIcon`.
Then the example here would be:
```suggestion
        error_message = ui.illustrated_message(
            ui.icon("vsWarning", size="XXL", margin_bottom="size-10"),
            ui.heading("Invalid Input"),
            ui.content("Please enter 'Sym' and 'Exchange' above"),
        )
```

Though ideally we should be passing those in as defaults when an `Icon` is passed into `IllustratedMessage`. We can capture that in a separate ticket though.

Reply:
<img width="937" alt="image" src="https://github.com/deephaven/deephaven-plugins/assets/55671206/80879764-5b14-4009-a460-61bc3f2dec18">

Unit tests failing because of the prop spreading